File: scripts/simple_action_client.py
#! /usr/bin/env python

import rospy
import actionlib

from lecture_pkg.msg import *

if __name__ == "__main__":
  rospy.init_node("do_dishes_client")
  client = actionlib.SimpleActionClient("do_dishes",DoDishesAction)
  client.wait_for_server()

  goal = DoDishesGoal()
  goal.dishwasher_id = 10
  print "Requesting dishwasher %d"%(goal.dishwasher_id)
  # Fill in ...
  client.send_goal(goal)
  print client.wait_for_result(rospy.Duration.from_sec(5.0))

  result = client.get_result()
  print "Resulting dishwasher %d"%(result.total_dishes_cleaned)

File: src/simple_action_client.cpp
#include <lecture_pkg/DoDishesAction.h>
#include <actionlib/client/simple_action_client.h>

typedef actionlib::SimpleActionClient<lecture_pkg::DoDishesAction> Client;
int main(int argc, char **argv)
{
    ros::init(argc, argv, "do_dishes_client");
    Client client("do_dishes", true); // true -> don't need ros::spin()
    client.waitForServer();
    lecture_pkg::DoDishesGoal goal;
    // Fill in goal here
    client.sendGoal(goal);
    client.waitForResult(ros::Duration(5.0));
    if(client.getState() == actionlib::SimpleClientGoalState::SUCCEEDED)
        printf("Yay! The dishes are now clean");
    printf("Current State: %s", client.getState().toString().c_str());
    return 0;
}
